### ISSUE TITLE
[Feature Request] use ttypescript under the hood to enable applying TS transforms

### ISSUE THREAD

**Issue by probablyup** [developit/microbundle#719]
https://www.npmjs.com/package/ttypescript

**Comment by probablyup** [comment]
Nvm, I see you're using https://github.com/ezolenko/rollup-plugin-typescript2. Will move the FR there.

**Comment by wardpeet** [comment]
First of all, thank you for opening a PR. Can you elaborate a little bit on why we want to allow typescript transformers? Why not go for a babel plugin?

I'm worried that if we swap typescript with a third party compiler, we will open up to more bugs that are not related to microbundle itself.

**Comment by berndartmueller** [comment]
I would also be in favor of including ttypescript so that I can use https://www.npmjs.com/package/ts-transformer-properties-rename to rename TS properties.

**Comment by waynevanson** [comment]
@wardpeet 

from https://github.com/cevek/ttypescript#what-it-is:

> Instead of tsc and tsserver, use ttsc and ttsserver wrappers. This wrappers try to use locally installed typescript first.
> 
> No version lock-ins - typescript used as peer dependency.

ttypescript uses typescript under the hood. I don't _BELEIVE_ anything should break. However, smoke tests should be written if no one else is certain.

**Comment by probablyup** [comment]
Yeah it basically just allows the rollup plugin to also look at tsconfig.json :)

**Comment by developit** [comment]
FWIW @berndartmueller Microbundle already provides that exact feature:

```jsonc
// package.json
{
  "mangle": {
    "properties": "^_"
  }
}
```

**Comment by timocov** [comment]
@developit I think `ts-transformer-properties-rename` can do more - it automatically detects what properties are "public" (or "external") and don't rename them, and what properties are internal (not just "private" in terms of class methods/properties, but all objects fields which aren't public/external) and rename them accordingly.

**Comment by Js-Brecht** [comment]
Maybe instead of hardcoding it to `ttypescript`, make a new cli option like `--typescript-compiler` (or `--tsc`?)?  Then it's an opt-in thing so if it breaks things, then people know it's probably not `microbundle`.
